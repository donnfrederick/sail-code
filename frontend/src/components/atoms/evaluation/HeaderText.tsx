import * as React from 'react'
import styled from 'styled-components'

interface ContainerProps {
  marginBottom?: number
  width?: number
}

interface Props extends ContainerProps {
  headerFontSize?: number
  subHeaderFontSize?: number
  texts: string[]
  subHeaderAlign?: 'left' | 'center'
}

export default ({
  width,
  headerFontSize,
  subHeaderFontSize,
  marginBottom,
  texts,
  subHeaderAlign
}: Props) => {
  const [header, subHeader] = texts
  return (
    <Container width={width} marginBottom={marginBottom}>
      <Heading fontSize={headerFontSize}>{header}</Heading>
      {subHeader && (
        <SubHeading fontSize={subHeaderFontSize} textAlign={subHeaderAlign}>
          {subHeader}
        </SubHeading>
      )}
    </Container>
  )
}

interface TextProps {
  fontSize?: number
}

interface SubHeadingProps extends TextProps {
  textAlign?: 'left' | 'center'
}

const Container = styled<ContainerProps, any>('div')`
  ${({ width }) => width && `width: ${width}px`};
  margin-bottom: ${({ marginBottom }) => (marginBottom ? marginBottom : 0)}px;
  display: inline-block;
`

const Heading = styled<TextProps, any>('h2')`
  margin: 0 0 30px;
  padding: 0;
  font-size: ${({ fontSize }) => (fontSize ? fontSize : 40)}px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  line-height: 1.5;
`

const SubHeading = styled<SubHeadingProps, any>('h3')`
  margin: 0 auto;
  padding: 0;
  font-size: ${({ fontSize }) => (fontSize ? fontSize : 32)}px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: ${({ textAlign }) => (textAlign ? textAlign : 'center')};
  line-height: 1.5;
  white-space: pre-wrap;
`
