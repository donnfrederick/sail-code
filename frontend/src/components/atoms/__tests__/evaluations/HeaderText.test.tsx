import HeaderText from 'components/atoms/evaluation/HeaderText'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <HeaderText />', () => {
  const tree = renderer
    .create(
      <HeaderText
        width={650}
        headerFontSize={40}
        subHeaderFontSize={32}
        marginBottom={20}
        texts={[
          'このかたの日本語レベルはどうdでしたか？',
          'どれか１つだけ選んでください。'
        ]}
        subHeaderAlign="left"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
