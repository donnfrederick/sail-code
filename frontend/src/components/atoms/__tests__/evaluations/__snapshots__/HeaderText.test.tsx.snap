// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <HeaderText /> 1`] = `
<div
  className="sc-bdVaJa FMzly"
  width={650}
>
  <h2
    className="sc-bwzfXH gxTqoK"
    fontSize={40}
  >
    このかたの日本語レベルはどうdでしたか？
  </h2>
  <h3
    className="sc-htpNat dVKQpm"
    fontSize={32}
  >
    どれか１つだけ選んでください。
  </h3>
</div>
`;
